src/bsp/platform/esp32/xi_esp32_bsp_component/xi_bsp_io_net_esp32.c

```diff
@@ -127,8 +127,10 @@ xi_bsp_io_net_state_t xi_bsp_io_net_write( xi_bsp_socket_t xi_socket,
 
     *out_written_count = write( xi_socket, buf, count );
 
-    if ( 0 > *out_written_count )
+    if ( *out_written_count < 0 )
     {
+        *out_written_count = 0;
```

Comment:
I just changed the `xi_io_net_layer.c` to close the whole connection if the BSP reports negative length bytes sent. This is treating negative length as an error. So this change is a fix in the BSP IO NET preventing confusion of the library.

Reply:
It may just be me, but this makes the BSP implementation more confusing. If negative values are used by sockets to indicate operational errors, and XI_BSP_IO_NET_STATE_ERROR is used in our BSP to indicate operational errors, why does the library also need `out_written_count`? Just as a sanity check?

Also, if this is a line that needs to be implemented identically in every BSP, shouldn't it be moved away from the BSP, and into the library? Something like `if ( XI_BSP_IO_NET_STATE_OK != bsp_state){ if (len<0){ xi_debug_log(...); len=0; }}`?

Reply:
Let me know if I'm missing something, or if I'm being too picky 😝

Reply:
You are right! Technically we don't need to set the `out_written_count` to zero if we return BSP error code since the `out_written_count` is ignored in error cases. It is just misleading to return negative bytes sent value, I mean it is weird, what would it mean? Returning zero in bytes sent variable is correct since in an error case nothing was sent, or at least the socket write API tells us, thus this is our best guess.
Other reason is the particular bug covered by this PR was the `XI_BSP_IO_NET_STATE_OK` return value and `out_written_count == -1` combination which made the library grow the buffer to send by appending 1 byte to the front in each loop.
Setting `out_written_count` to zero adds another guard against such scenario. And makes the function returning correct values.
But again technically we can get rid of this zero assignment.

Reply:
Ahh, I see. That's a good point. Looks good, then :+1: Thanks for clearing that up :)